### ISSUE TITLE
Type aware rules crash on new .vue files(after config is loaded)

### ISSUE THREAD

**Issue by Dtsiantaris** [vuejs/eslint-config-typescript#316]
### Environment

| | |
|---|---|
| `@vue/eslint-config-typescript` | 14.7.0 |
| `eslint` | 10.0.0 |
| `typescript-eslint` | 8.56.0 |
| `vue-eslint-parser` | 10.4.0 |
| `typescript` | 5.9.3 |

### What happens

Any `.vue` file created *after* the ESLint config module was evaluated fails to lint, in any long-lived process (`vite dev` + `vite-plugin-checker`, the VS Code ESLint server, `eslint --watch`):

```
Error while loading rule '@typescript-eslint/no-array-delete': You have used a rule which
requires type information, but don't have parserOptions set to generate type information
for this file.
Parser: vue-eslint-parser
```

`no-array-delete` is just first alphabetically — it's every type-aware rule. Because it throws at rule-load, the whole lint run for that batch aborts, so nothing is reported at all. A fresh `eslint <newfile>.vue` passes, which is why "restart and it goes away" masks this.

### Root cause

`insertAndReorderConfigs()` calls `groupVueFiles()` **once**, when the config module is evaluated. It's a synchronous `fast-glob` scan whose result is baked into literal `files:` arrays.

That creates an asymmetry between rules and parser:

- Type-aware **rules** apply to *all* `.vue` files — `TsEslintConfigForVue.toConfigArray()` rewrites any config whose `files` includes `**/*.ts` into `[...files, '**/*.vue']`.
- The **parser + `projectService`** applies only to the enumerated snapshot — `@vue/typescript/default-project-service-for-vue-files` gets `files: typeCheckableVueFiles.map(escapePathForGlob)`.

A file created after the scan is in neither `typeCheckable` nor `nonTypeCheckable`, so it keeps the plain `vue-eslint-parser` from `@vue/typescript/setup` while type-aware rules stay enabled → `getParserServices()` throws.

Note this cannot be worked around by re-importing the config: ESLint keys its config cache on the `eslint.config.js` mtime, and the package's own module instance stays cached, so the top-level `defineConfigWithVueTs()` call never re-runs. Only a full process restart re-scans.

### Reproduction

```js
import { ESLint } from 'eslint'
import fs from 'node:fs'

const eslint = new ESLint({ cwd: process.cwd() })
// Snapshot is taken here, before the file exists.
await eslint.calculateConfigForFile('src/components/Existing.vue')

const p = 'src/components/CreatedLater.vue'
fs.writeFileSync(p, '<template>\n  <div />\n</template>\n\n<script setup lang="ts">\n</script>\n')
try {
  await eslint.lintFiles([p])   // throws
} finally {
  fs.unlinkSync(p)
}
```

Config is `defineConfigWithVueTs(pluginVue.configs['flat/recommended'], js.configs.recommended, vueTsConfigs.recommendedTypeChecked)`.

### Expected

A `.vue` file created after config load should lint like any other, or at minimum not abort the run.

### Workaround I did

Confirmed working as a downstream workaround — appended after `vueTsConfigs.recommendedTypeChecked`:

```js
{
  files: ['**/*.vue'],
  languageOptions: {
    parserOptions: {
      projectService: true,
      extraFileExtensions: ['.vue']
    }
  }
}
```

### Verdict

Probably related to #52 and maybe other issues spread around various repos that tackle type aware rules.
